File: webserver/src/main/kotlin/cz/vutbr/fit/knot/enticing/webserver/entity/ipaddress/IpAddressCollection.java
package cz.vutbr.fit.knot.enticing.webserver.entity.ipaddress;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({FIELD})
@Retention(RUNTIME)
@Constraint(validatedBy = IpAddressCollectionValidator.class)
@Documented
public @interface IpAddressCollection {
    String message() default "Invalid Ip address collection";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: webserver/src/main/kotlin/cz/vutbr/fit/knot/enticing/webserver/entity/validation/IpAddressCollection.kt
package cz.vutbr.fit.knot.enticing.webserver.entity.validation

import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Constraint(validatedBy = [IpAddressCollectionValidator::class])
annotation class IpAddressCollection(
        val message: String = "Invalid Ip address collection",
        val groups: Array<KClass<Any>> = [],
        val payload: Array<KClass<Payload>> = []
)